src/System.CommandLine/Argument.cs

```diff
@@ -65,6 +66,16 @@ public Argument(string name)
             set => _arity = value;
         }
 
+
+        /// <summary>
+        /// Gets or sets if the ArgumentHelpName is specified by the user.
+        /// </summary>
+        public bool IsHelpNameSpecified
```

Comment:
This should probably be `internal`.

Reply:
Actually, this feels like an implementation concern of the `HelpBuilder` rather than something intrinsic to `Argument`. Maybe just move the check there e.g. `string.IsNullOrEmpty(argument.Name)`?

Reply:
I can make it internal. Although I don't think we can just check `string.IsNullOrEmpty(argument.Name)` since the argument name gets a default value when not set explicitly: https://github.com/dotnet/command-line-api/blob/e0edf1801e2ba9b4b5f733e3484aa7dbe8bc8aca/src/System.CommandLine/Argument.cs#L124

Reply:
Sorry, I was thinking of `ArgumentHelpName` and realized that that's only present on `Option`. It might be useful to decouple this from `Argument.Name`.

Reply:
How so, add a separate help name field on the argument class?

Reply:
Yeah, I think that's nice and simple. @Keboo?

Reply:
LGTM

Reply:
Just pushed this change.